src/content/reference/react/cache.md

```diff
@@ -0,0 +1,443 @@
+canary: true
+---
+
+<Canary>
+`cache` is currently only available in React’s [Canary](https://react.dev/community/versioning-policy#canary-channel) and [experimental](https://react.dev/community/versioning-policy#experimental-channel) channels. Please ensure you understand the limitations before using `cache` in production.
+
+Learn more about [React's release channels here](/community/versioning-policy#all-release-channels).
+
+</Canary>
+
+<Intro>
+
+`cache` lets you cache the result of a data fetch or computation.
+
+```js
+const cachedFn = cache(fn);
+```
+
+</Intro>
+
+<InlineToc />
+
+---
+
+## Reference {/*reference*/}
+
+### `cache(fn)` {/*cache*/}
+
+Wrap a function with `cache` to receive a version of that function with caching.
+
+```js {4,7}
+import {cache} from 'react';
+import fetchUser from 'lib/user';
+
+const getUser = cache(fetchUser);
+
+function UserProfile(id) {
+  const user = getUser(id);
+  //...
+}
+```
+
+`getUser` will first check if there is a cached result for `id`, otherwise call `fetchUser(id)` and cache the returned result.
+
+[See more examples below.](#usage)
+
+#### Parameters {/*parameters*/}
+
+- `fn`: The function you want to cache results for. `fn` can take any arguments and return any value.
+
+#### Returns {/*returns*/}
+
+`cache` returns a cached version of `fn` with the same type signature.
+
+When calling `cachedFn` with given arguments, it first checks if a cached result exists in the cache. If a cached result exists, it returns the result. If not, it calls `fn` with the arguments, stores the result in the cache, and returns the result.
+
+<Note>
+
+The optimization of caching return values based on inputs is known as [_memoization_](https://en.wikipedia.org/wiki/Memoization). We refer to `cachedFn` as a memoized function.
+
+</Note>
+
+#### Caveats {/*caveats*/}
+
+[//]: # 'TODO: add links to Server/Client Component reference once https://github.com/reactjs/react.dev/pull/6177 is merged'
+
+- `cache` is recommended for Server Components only. There are plans to introduce `cache` for Client Components, but it is not recommended today.
+- React will invalidate the cache for a memoized function across server requests.
+- Each call to `cache` creates a new function. This means that wrapping `cache` around the same function multiple times will return different memoized functions and they will not share the same cache.
+- The benefit of `cache` is to skip duplicate work by sharing a cache. To promote cache sharing, `cachedFn`, should be defined in a scope that is accessible to multiple components. In most cases, this means calling `cache` and defining the memoized function in the global scope.
+- Cache access only occurs during a component render. This means a call to `cachedFn` outside of a Server Component will not update or read the cache.
+- `cachedFn` will also cache errors. If `fn` throws an error for certain arguments, it will be cached, and re-thrown when `cachedFn` is called with those arguments.
```

Comment:
Will it re-throw with the original callstack or the callstack that's specific to the current call?

Reply:
Good question -- it looks like the original callstack: https://github.com/facebook/react/blob/456d153bb582798effa76c09bec2405ab2e392cf/packages/react/src/ReactCache.js#L122

Reply:
Clarified that this is the same error re-thrown